### PR TITLE
Add `drill` integration tests

### LINKED ISSUE #38578: Add `drill` test integration
### Body

During https://github.com/apache/airflow/pull/38530 we found that for a long time (since initial commit to the provider TBH) configured to invalid DB Backend.  Unit test which moved into the integration tests:
- `tests/providers/apache/drill/operators/test_drill.py` -> [`tests/integration/providers/apache/drill/operators/test_drill.py`](https://github.com/apache/airflow/blob/main/tests/integration/providers/apache/drill/operators/test_drill.py)

We should eventually add Apache Drill as available option for integration tests, and make to possible to run integration tests with it

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by shahar1**
closes: #38578

This sets the integration env. and adds a basic test to the `drill`'s hook.
We could add more tests later :)

CC: @potiuk, @eladkal

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/global_constants.py [hunk 1]
```diff
@@ -60,6 +60,7 @@
     "kafka",
     "qdrant",
     "mssql",
+    "drill",
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/global_constants.py]: NIT: can we sort these alphabetically :D  ? It hurts a bit looking at the list being semi-random order.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
LGTM - just one, really small, NIT

**Review by eladkal — APPROVED** [review]

File: dev/breeze/src/airflow_breeze/global_constants.py (hunk 1 above)
Comment (shahar1) [reply dev/breeze/src/airflow_breeze/global_constants.py]: Fixed, please don't take a look at the rest of the file :)

**Comment by shahar1** [comment]
The currently failed test seems unrelated

File: dev/breeze/src/airflow_breeze/global_constants.py (hunk 1 above)
Comment (potiuk) [reply dev/breeze/src/airflow_breeze/global_constants.py]: My approach is to fix and sort what I touch :). If you look at some of my changes from last months many of them actually did some sorting. And the thing is ....

Let's make every change leave our small "world" in a slightly better shape than what we got before - **sorting out** one thing at a time :D.

**Pull request merged**